lib/vtls/vtls_scache.c

```diff
@@ -359,7 +359,7 @@ static CURLcode cf_ssl_peer_key_add_path(struct dynbuf *buf,
     char abspath[_MAX_PATH];
     if(_fullpath(abspath, path, _MAX_PATH))
       return Curl_dyn_addf(buf, ":%s-%s", name, abspath);
-#else
+#elif !defined(__OS400__)
```

Comment:
I propose we instead add a check for realpath to cmake + configure and use `#ifndef HAVE_REALPATH` here, to avoid having the code to know which specific systems that lack it.

Reply:
Here's my alternative take:  #16225